service/history/api/create_workflow_util.go

```diff
@@ -60,14 +60,14 @@ type (
 	}
 )
 
-func NewWorkflowWithSignal(
+func NewWorkflow(
 	shard shard.Context,
 	namespaceEntry *namespace.Namespace,
 	workflowID string,
 	runID string,
 	startRequest *historyservice.StartWorkflowExecutionRequest,
-	signalWithStartRequest *workflowservice.SignalWithStartWorkflowExecutionRequest,
-) (WorkflowLease, error) {
+	beforeWorkflowTaskStartedEventHook func(workflow.MutableState) error,
+) (workflow.MutableState, error) {
```

Comment:
Returning only the MS instead of the lease allows the caller to construct the lease.